distsql/distsql_test.go

```diff
@@ -500,50 +430,135 @@ func createSelectNormal(batch, totalRows int, ctx sessionctx.Context) (*selectRe
 	return result, colTypes
 }
 
-func BenchmarkSelectResponseChunk_BigResponse(b *testing.B) {
-	for i := 0; i < b.N; i++ {
-		b.StopTimer()
-		s := &testSuite{}
-		s.SetUpSuite(nil)
-		s.sctx.GetSessionVars().InitChunkSize = 32
-		s.sctx.GetSessionVars().MaxChunkSize = 1024
-		selectResult, colTypes := createSelectNormal(4000, 20000, s.sctx)
-		chk := chunk.NewChunkWithCapacity(colTypes, 1024)
-		b.StartTimer()
-		for {
-			err := selectResult.Next(context.TODO(), chk)
-			if err != nil {
-				panic(err)
-			}
-			if chk.NumRows() == 0 {
-				break
-			}
-			chk.Reset()
-		}
-		s.TearDownSuite(nil)
+func testChunkSize(t *testing.T, response SelectResult, colTypes []*types.FieldType) {
+	chk := chunk.New(colTypes, 32, 32)
+
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+
+	chk.SetRequiredRows(1, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 1, chk.NumRows())
+
+	chk.SetRequiredRows(2, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 2, chk.NumRows())
+
+	chk.SetRequiredRows(17, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 17, chk.NumRows())
+
+	chk.SetRequiredRows(170, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+
+	chk.SetRequiredRows(32, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+
+	chk.SetRequiredRows(0, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+
+	chk.SetRequiredRows(-1, 32)
+	require.NoError(t, response.Next(context.TODO(), chk))
+	require.Equal(t, 32, chk.NumRows())
+}
+
+func createSelectNormal(t *testing.T, batch, totalRows int, planIDs []int, sctx sessionctx.Context) (*selectResult, []*types.FieldType) {
```

Comment:
Almost lgtm.
According to the habit of `Go`, the `sctx` seems should not be the last argument. How about move it following `t *testing.T`.  What's your opinion? @tisonkun

Reply:
Make sense.

BTW I think these tests are written complex, we may later try to simplify them. For example, pass `nil` in `createSelectNormal` is do if-else if not a good practice. Also if we have a dedicated test for create request and so on, we don't have to pass `testing.T` here. But let's postpone to followups.